inc/dbmysql.class.php

```diff
@@ -381,7 +381,12 @@ function field_flags($result, $field) {
     * @return list of tables
    **/
    function list_tables($table="glpi_%") {
-      return $this->query("SHOW TABLES LIKE '".$table."'");
+      return $this->query(
+         "SELECT TABLE_NAME FROM information_schema.`TABLES`
+             WHERE TABLE_SCHEMA = '{$this->dbdefault}'
+                AND TABLE_TYPE != 'VIEW'
```

Comment:
So you may change it to the positive test, so that if tomorrow there is something else it will not be broken, as the meaning of it is to get the tables, and not to get something that is not the views :)